subctl/issue-89-reconstructing-broker-info.md

```diff
@@ -0,0 +1,62 @@
+# Reconstruct broker-info.subm file from the running Broker
+
+Related Issue:
+[Option for recovering the broker-info.subm file from a running broker cluster](https://github.com/submariner-io/enhancements/issues/143)
+
+## Summary
+
+`broker-info.subm` file is created when a broker is installed. The file contains information critical to have a functional environment. 
+There is a possibility that the file could be deleted or lost for one or the other reason. In such cases, the environment becomes redundant
+and no new clusters could be joined.
+
+This proposal describes a way to reconstruct the `broker-info.subm` file from the running broker.
+
+## Proposal
+
+A new `subctl` command will be added that would fetch needed data from the broker and write it to `broker-info.subm` file. The file 
+would be Base64 encoded.
+
+## Design details
+
+* subctl command name: `reconstruct-brokerinfo`.
+* Flags for the command:
+  * `namespace` to specify the namespace in which broker is installed.
+  * `ipsec-psk-from` to specify the file to read PSK secret from.
+  * All flags that `restConfigProducer` provides.
+* Data to be fetched from running broker:
+  * Broker URL
+  * Client token
+  * IPSec PSK
+  * Components installed i.e. Submariner and Service discovery
+  * whether service discovery component is installed
+  * default custom domains
+
+The last three information will be retrieved from the broker `Spec`, Broker URL from the user provided kubeconfig, client token from the 
+Broker admin service account `submariner-k8s-broker-admin` and IPSec PSK with the following logic:
+
+```
+if file specified via `ipsec-psk-from` flag by the user
+  read the secret from the specified file
+else if submariner is installed in the cluster
+  read the secret from submariner spec
+else
```

Comment:
There’s a missing case here: if the user didn’t provide a file, and Submariner isn’t installed (or isn’t accessible using the available credentials), but the broker has one or more connected clusters, `subctl` should fail with an error — a `broker-info.subm` with a new PSK would not allow newly-connected clusters to connect to existing clusters.

Reply:
Done.